packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+import Foundation
+import EmbraceIO
+import EmbraceCore
+import EmbraceCrash
+import EmbraceCommonInternal
+
+#if canImport(CodePush)
+import CodePush
+#endif
+
+enum EmbraceKeys: String {
+  case reactNativeVersion = "io.embrace.reactnative.version"
+  case embraceReactNativeSdkVersion = "io.embrace.reactnative.sdk.version"
+  case javaScriptPatchNumber = "io.embrace.javascript.patch"
+  case javaScriptBundleURL = "io.embrace.jsbundle.url"
+}
+
+@objc(EmbraceManager)
+class EmbraceManager: NSObject {
+  @objc
+  func setJavaScriptBundlePath(_ path: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
```

Comment:
for this and the other methods I think `@escaping` is only required when using the variables in a closer, are they needed here because of the `do...catch`? Are they needed in other methods that don't include that?

Reply:
Since we are using promise in RN, it's good practice to use @escaping when you work with async functions. Even though the content is not directly async right now. 
If the content is not async, you could remove it. Let me know if you want me to remove it ( I would keep it)

Reply:
ok let's keep it